dockhub.py

```diff
@@ -1,226 +1,182 @@
 #!/usr/bin/env python3
 
+from functools import wraps
 import json
 from os import getenv
-from sys import exit
+import sys
 
 import click
 import requests
 
 
-def die(msg=None, exception=None):
-    if msg is not None:
-        print(f'{msg}')
-    if exception is not None:
-        print(f'{exception}')
-    exit(1)
+CONTENT_HEADER = {'Content-Type': 'application/json', 'charset': 'utf-8'}
+DOCKERHUB_URL = 'https://hub.docker.com/v2'
 
 
+def handle_http_errors(fun):
+    """Wrap function with exception handling for common HTTP errors."""
+    @wraps(fun)
+    def _handle_http_errors(*args, **kwargs):
+        try:
+            return fun(*args, **kwargs)
+
+        except requests.ConnectionError as exc:
+            raise click.ClickException(f'Unable to connect to dockerhub: {exc}')
```

Comment:
click has a `ClickException` from which all its other exceptions derive. I try to use it when I can since click will print out helpful things and does the right exit code.